environment.yml

```diff
@@ -117,3 +117,4 @@ dependencies:
 
   - pip:
       - sphinx-toggleprompt
+      - typing_extensions; python_version<"3.11"
```

Comment:
I think we don't need to add it, as it is only used during type checking (and typing_extensions is, I think, a dependency of mypy).

Reply:
Yes, typing_extension is a dependency of mypy, I just checked for that. We do not want to be explicit, now that we're importing fro typing_extensions?

Reply:
I think even the stubs packages from typeshed (for example https://pypi.org/project/types-psutil/) do not have typing_extensions as a dependency (neither does pandas-stubs).

If it is a new dependency, it probably should be added to the whatsnew (there might be other places where it needs to be added as well).

Reply:
we have had discussions about this before and the outcome was that we do not use typing_extensions.

IIUC we don't need it with the latest mypy (1.0.0) when checking with python 3.11